### PR TITLE
[runtime] Fix cross-platform bitcode option auto-detection

### PR DESCRIPTION
**Pull request by alexanderkyte**
@monojenkins skip

### CODE REVIEW

**Comment by alexanderkyte** [comment]
@monojenkins skip

File: scripts/ci/run-jenkins.sh [hunk 1]
```diff
@@ -64,7 +64,7 @@ elif [[ ${CI_TAGS} == *'fullaot'* ]];            then EXTRA_CONF_FLAGS="${EXTRA_
 elif [[ ${CI_TAGS} == *'hybridaot'* ]];          then EXTRA_CONF_FLAGS="${EXTRA_CONF_FLAGS} --with-runtime-preset=hybridaot";
 elif [[ ${CI_TAGS} == *'winaot'* ]];             then EXTRA_CONF_FLAGS="${EXTRA_CONF_FLAGS} --with-runtime-preset=winaot";
 elif [[ ${CI_TAGS} == *'aot'* ]];                then EXTRA_CONF_FLAGS="${EXTRA_CONF_FLAGS} --with-runtime-preset=aot";
-elif [[ ${CI_TAGS} == *'bitcode'* ]];            then EXTRA_CONF_FLAGS="${EXTRA_CONF_FLAGS} --with-runtime-preset=bitcode"; export MONO_ENV_OPTIONS="$MONO_ENV_OPTIONS --aot=clangxx=clang++-6.0";
+elif [[ ${CI_TAGS} == *'bitcode'* ]];            then EXTRA_CONF_FLAGS="${EXTRA_CONF_FLAGS} --with-runtime-preset=bitcode --with-aot-clang-path=clang++-6.0";
```
Comment (vargaz) [inline scripts/ci/run-jenkins.sh]: Whats wrong with the previous version ?

Comment (alexanderkyte) [reply scripts/ci/run-jenkins.sh]: That option is applied everywhere so when it tries to execute code that has been aot'ed, it ends up also passing `--aot=` . It causes the profile check to fail, the mcs portion of the build terminates immediately.

Comment (alexanderkyte) [reply scripts/ci/run-jenkins.sh]: This change makes us only pass it when we're compiling the AOT code

**Pull request closed without merging**